src/AutoRest.CSharp/Common/Generation/LongRunningOperationWriter.cs

```diff
@@ -70,13 +70,21 @@ void WriteResultFunction(bool async)
             var @namespace = cs.Namespace;
             using (writer.Namespace(@namespace))
             {
+                bool isGenericOperation = !(operation.ResultType.IsFrameworkType && operation.ResultType.FrameworkType == typeof(Response));
```

Comment:
We should fix the logic in https://github.com/Azure/autorest.csharp/blob/feature/v3/src/AutoRest.CSharp/Common/Output/Models/LongRunningOperation.cs#L52 instead.